### PR TITLE
Grammar change to aid readability. Also added hyperlink to styled-com…

### PR DESCRIPTION
**Pull request by Ekwuno**
…ponent plugin

## Description

While reading through the Doc for plugins, I saw that the introduction statement for the second paragraph could be a bit more detailed  😄

### Documentation

  Where is this feature or API documented?
     https://www.gatsbyjs.org/docs/what-you-dont-need-plugins-for/
   
 

## Related Issues

  Link to related issues 
  e.g. Related to #19277

### CODE REVIEW

File: docs/docs/what-you-dont-need-plugins-for.md [hunk 1]
```diff
@@ -9,4 +9,6 @@ Some examples:
 - Importing JavaScript packages that provide general functionality, such as `lodash` or `axios`
 - Integrating visualization libraries, such as `Highcharts` or `d3`.
 
-As a general rule, you may use _any_ npm package you might use without Gatsby, with Gatsby. What plugins offer is a prepackaged integration into the core Gatsby APIs to save you time and energy, with minimal configuration. In the case of `Styled Components`, you could manually render the `Provider` component near the root of your application, or you could just use `gatsby-plugin-styled-components` which takes care of this step for you in addition to any other difficulties you may run into configuring Styled Components to work with server side rendering.
+As a general rule, you may use any npm package you might use while working on a JavaScript or React application ,when deciding to work with Gatsby. What plugins offer is a prepackaged integration into the core Gatsby APIs to save you time and energy, with minimal configuration.
```
Comment (laurieontech) [inline docs/docs/what-you-dont-need-plugins-for.md]:
```suggestion
As a general rule, any npm package you might use while working on another JavaScript or React application can also be used with a Gatsby application. What plugins offer is a prepackaged integration into the core Gatsby APIs to save you time and energy, with minimal configuration.
```

File: docs/docs/what-you-dont-need-plugins-for.md [hunk 2]
```diff
@@ -9,4 +9,6 @@ Some examples:
...
+
+A good use case would be using `Styled Components`, you could manually render the `Provider` component near the root of your application, or you could just use [gatsby-plugin-styled-components ](https://www.gatsbyjs.org/packages/gatsby-plugin-styled-components/)which takes care of this step for you in addition to any other difficulties you may run into configuring Styled Components to work with server side rendering.
```
Comment (laurieontech) [inline docs/docs/what-you-dont-need-plugins-for.md]:
```suggestion
A good use case would be using `Styled Components`, you could manually render the `Provider` component near the root of your application, or you could use [gatsby-plugin-styled-components](/packages/gatsby-plugin-styled-components/) which takes care of this step for you in addition to any other difficulties you may run into when configuring Styled Components to work with server-side rendering.
```

File: docs/docs/what-you-dont-need-plugins-for.md (hunk 1 above)
Comment (Ekwuno) [reply docs/docs/what-you-dont-need-plugins-for.md]: I removed the yarn.lock from the commit

**Comment by marcysutton** [comment]
The `yarn.lock` file is being a little stubborn, I see! It does exist in the repo, so it shouldn't be deleted as a file, just removed from the PR. I merged from master manually and the file should be there, but it's still showing as deleted on GitHub...it may be easier to start with a new PR if it's still sticking around as a deleted file. 

Thanks for the contribution, @Ekwuno!!

**Comment by Ekwuno** [comment]
Mikhail gave me a fix I am going to try that out. 🤞🏽 maybe it works.

**Comment by Ekwuno** [comment]
It didn't I will open a new PR for this. Thanks.

**Comment by sidharthachatterjee** [comment]
> It didn't I will open a new PR for this

Alright, thanks! 🙌

**Pull request closed without merging**